### PR TITLE
Optimize database queries

### LINKED ISSUE #161: Do not use .exists()
I see a lot of examples in the code when you use `.exists` method. 

For example:

```python
user_check = User.objects.filter(
    email__iexact=data.get('email').strip()
)
if user_check.exists():
    try:
        user_profile = UserProfile.objects.get(
            user=user_check.last()
        )
```

Please, do not use this method if you plan to get data in the future. When you run it you execute 2 database queries instead of only one. If user object doesn't exist you will get an empty DB query that cost nothing for the DB, but if a user exists you will get only query instead of two. For example this part of a code you can rewrite like this: 

```python
user_check = User.objects.filter(
    email__iexact=data.get('email').strip()
).last() #  None if there is no user
# and here all the verification you have 
```
Also in this part you can see that you execute two query for the User model and one for user profile, but you can make only one like this: 
```sql 
SELECT "venue_userprofile"."email_confirmed"
FROM "auth_user"
       LEFT JOIN "venue_userprofile" ON ("auth_user"."id" = "venue_userprofile"."user_id")
WHERE UPPER("auth_user"."email" :: TEXT) = UPPER(SOME email)
```
Using ORM query like this one: 
```python
email_confirmed = User.objects.filter(
     email__iexact=data.get('email').strip()
).values('profiles__email_confirmed').last()
response['email_exists'] = email_confirmed['profiles__email_confirmed'] if email_confirmed else False
```

P.S. `exists` works very well but when you don't want to use data from the query(only check if exists).

### PR DESCRIPTION
**Pull request by joemarct**
This closes #161. Other commits that attempt to optimize DB queries may be added here.

### CODE REVIEW

**Review by shawnlauzon — CHANGES_REQUESTED** [review]
Seems we should be calling `lower` every time we call `strip`.

File: venue/views.py [hunk 1]
```diff
@@ -351,7 +351,11 @@ def create_user(request):
     """
     data = request.data
     data['email'] = data['email'].lower()
-    user_check = User.objects.filter(email=data['email'])
+    # Check if a user with confirmed email same as the given already exists
+    user_check = UserProfile.objects.filter(
+        user__email__iexact=data['email'].strip(),
```
[the remark this answers is not in the document]
Comment (joemarct) [reply venue/views.py]: `user__email__iexact` --> The `__iexact` filter here means case insensitive matching, so I think `.lower()` is redundant.

File: venue/views.py [hunk 2]
```diff
@@ -1616,7 +1613,7 @@ def check_username_exists(request):
     data = request.query_params
     response = {'success': True, 'username_exists': False}
     if data.get('username'):
-        user_check = User.objects.filter(username=data.get('username').lower())
+        user_check = User.objects.filter(username__iexact=data.get('username').strip())
```
[the remark this answers is not in the document]
Comment (joemarct) [reply venue/views.py]: That's why I changed the query to `username__iexact` for case-insensitive matching, same effect as `.lower()`.

File: venue/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shawnlauzon) [reply venue/views.py]: Ahhh I didn't realize that. Ok, that sounds like it works then.

**Review by shawnlauzon — APPROVED** [review]

**Pull request merged**